packages/gatsby-theme-api-docs/src/components/resource/resource-by-key.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useReadResourceByResourcePath } from '../../hooks/use-read-resource-by-resource-path';
+import Resource from './resource';
+import reportError from '../../utils/report-error';
+
+const ResourceByKey = ({ apiKey, resource }) => {
+  const resourceObj = useReadResourceByResourcePath(apiKey, resource);
+
+  if (!resourceObj) {
+    return reportError(`Resource '${resource}' not found in '${apiKey}' API`);
+  }
+
+  return <Resource resourceObj={resourceObj} />;
+};
+
+ResourceByKey.propTypes = {
+  apiKey: PropTypes.string.isRequired,
+  resource: PropTypes.string.isRequired,
+};
+export default ResourceByKey;
```

Comment:
Please what is the benefit of wrapping with `< ResourceByKey />`?

Reply:
See description text of the PR.  (enable using the child component in a custom wrapper that uses a hook that loads less or different data)